src/screens/RegistrationScreen/DonorRegistrationScreen.tsx

```diff
@@ -22,100 +23,236 @@ import { getStateList } from '@util/statesAbbr';
 import styles from './RegistrationScreen.styles';
 
 export default () => {
-	const { navigate } = useNavigation();
+	const { navigate, goBack } = useNavigation();
 	const [ _state, actions ] = useGlobal() as any;
 	const { register } = actions;
 
-	const [ city, setCity ] = useState('');
 	const [ email, setEmail ] = useState('');
-	const [ image, setImage ] = useState({} as ImageInfo);
-	const [ license, setLicense ] = useState('');
-	const [ organizationName, setOrganizationName ] = useState('');
 	const [ password, setPassword ] = useState('');
-	const [ street, setStreet ] = useState('');
+	const [ retypedPassword, setRetypedPassword ] = useState('');
+	const [ firstName, setFirstName ] = useState('');
+	const [ lastName, setLastName ] = useState('');
+	const [ businessName, setBusinessName ] = useState('');
+	const [ address, setAddress ] = useState('');
+	const [ city, setCity ] = useState('');
 	const [ state, setState ] = useState('WA');
-	const [ termsOfService, setTermsOfService ] = useState(false);
 	const [ zip, setZip ] = useState('');
+	const [ instructions, setInstructions ] = useState('');
+	const [ termsOfService, setTermsOfService ] = useState(false);
 	const stateList = getStateList();
 
+	// object contains the error status and corresponding message
+	// modifications in errorObj trigger change in UI.
+	const errorObj = {
+		email: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		password: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		retypedPassword: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		firstName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		lastName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		businessName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		address: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		city: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		state: {
```

Comment:
Same sort of questions here as I had with ClientRegistration.  Have you looked at something like https://react-hook-form.com/ for form validations?  I'm not sure if that's practical, but ideally we'd use a library for form validation.

Reply:
That library would be good, but we already have validate.js and it would validate the content in register function. Not sure if it's necessary to replace it.